packages/jest-cli/src/TestRunner.js

```diff
@@ -264,16 +268,13 @@ class TestRunner {
       this.addReporter(new NotifyReporter());
     }
 
-    if (this._config.collectCoverage) {
-      const CoverageReporter = require('./reporters/CoverageReporter');
-      this.addReporter(new CoverageReporter());
-    }
   }
 
   _bailIfNeeded(aggregatedResults: AggregatedResult) {
     if (this._config.bail) {
-      this._dispatcher.onRunComplete(this._config, aggregatedResults);
-      process.exit(1);
+      this._dispatcher
+        .onRunComplete(this._config, aggregatedResults)
+        .then(() => process.exit(1));
```

Comment:
I don't think this is how it works. In bail mode this would continue running tests until the reporters are done running.